p_sets/3/find.rb

```diff
@@ -0,0 +1,56 @@
+module Finder
+  def self.linear_search(needle, hay)
+    max_index   = hay.length - 1
+    result      = false
+    max_index.times do |i|
+      result = true if hay[i] == needle
+    end
+    puts result.to_s
+  end
+
+  def self.binary_search(needle, hay)
+    mid_point = (hay.length - 1) / 2
+    max_point = hay.length - 1
+    left_side = 0..mid_point
+    right_side = (mid_point + 1)..max_point
+
+    if needle < hay[0] || needle > hay[max_point]
+      result = false
+    else
+      if needle < hay[mid_point]
+        binary_search(needle, hay[left_side])
+      elsif needle > hay[mid_point]
+        binary_search(needle, hay[right_side])
+      else
+        result = hay[mid_point]
```

Comment:
I don't think my binary search works as it should, isn't my midpoint always going to be 0 unless on the first pass through the needle happens to be at the midpoint? I've been looking at this example:
```
def binary_search(array, value, from=0, to=nil)
    if to == nil
        to = array.count - 1
    end

    mid = (from + to) / 2

    if value < array[mid]
        return binary_search array, value, from, mid - 1
    elsif value > array[mid]
        return binary_search array, value, mid + 1, to
    else
        return mid
    end
end
```
and I can't quite follow why this one returns the mid from the original array, but mine returns the mid from the final array.